MekHQ/src/mekhq/campaign/Campaign.java

```diff
@@ -7172,6 +7192,41 @@ public void assignRandomPortraitFor(Person p) {
                 }
             }
         }
+
+        if (possiblePortraits.isEmpty()) {
```

Comment:
This if statement block seems like a prime candidate to split out into a separate function, as it recurs three times and is mostly identical.